File: 프로그래머스/레벨2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def compare(x, y):
    left = x + y
    right = y + x
    
    if left > right:
        return -1
    elif left < right:
        return 1
    else:
        return 0

def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=cmp_to_key(compare))
    return "0" if numbers[0] == "0" else "".join(numbers)

File: 프로그래머스/레벨2/42746. 가장 큰 수/가장 큰 수.js
const compare = (x, y) => {
    const left = x + y;
    const right = y + x;
    
    if (left > right) return -1;
    else if (left < right) return 1;
    else return 0;
}

function solution(numbers) {
    const result = numbers.map(String).sort(compare);
    return result[0] === "0" ? "0" : result.join("");
}